### PR TITLE
[@bramus/pagination-sequence] Add a definition

### PR DESCRIPTION
**Pull request by kristjanjokull**
Add new definition for [https://github.com/bramus/js-pagination-sequence](https://github.com/bramus/js-pagination-sequence)

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/bramus__pagination-sequence/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+interface Options {
+    curPage?: number;
+    numPages?: number;
+    numPagesAtEdges?: number;
+    numPagesAroundCurrent?: number;
+    glue?: string;
+}
+
+declare function generateFromObj(opts: Options): Array<string | number>;
```
Comment (peterblazejewicz) [inline types/bramus__pagination-sequence/index.d.ts:22]: this one should take optional parameter IMO, as source uses default parameter value, readme:
https://github.com/bramus/js-pagination-sequence/tree/main#generatefromobj
```suggestion
declare function generateFromObj(opts?: Options): Array<string | number>;
```

**Comment by kristjanjoksig** [comment]
@peterblazejewicz I've now updated the generateFromObj parameter, so now it's optional.

**Comment by andrewbranch** [comment]
The upstream package here is invalid due to shipping CJS files with `.js` extensions under `"type": "module"` in its package.json. Let's see if https://github.com/bramus/js-pagination-sequence/pull/7 goes anywhere first. Otherwise, it will be kind of impossible to write a valid configuration for these types, since it represents an invalid runtime package configuration.

**Comment by bramus** [comment]
(Apologies for the late reply, I was OOO the past two weeks)

Thank you for contributing this @kristjanjokull. I have merged the PR by @andrewbranch to fix the package exports and will publish a new version soon.

**Comment by peterblazejewicz** [comment]
note: waits for upstream

**Comment by bramus** [comment]
[Version 1.2.1 of `@bramus/pagination-sequence`](https://www.npmjs.com/package/@bramus/pagination-sequence/v/1.2.1) just got published. It includes the required changes.

**Comment by kristjanjokull** [comment]
Ready to merge? @peterblazejewicz @andrewbranch @bramus

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@kristjanjoksig thanks!

**Comment by kristjanjoksig** [comment]
Ready to merge

**Comment by kristjanjoksig** [comment]
Ready to merge

**Comment by kristjanjokull** [comment]
Ready to merge

**Pull request merged**